Jenkinsfile

```diff
@@ -32,47 +31,65 @@ def conda_packages = [
     "verhawk",
     "pytest"
 ]
-def conda_packages_docs = [
-    "sphinx",
-    "sphinx_rtd_theme",
-    "stsci_rtd_theme"
-]
 
 // Pip related setup
-def pip_packages_docs = "sphinx-automodapi"
+def pip_packages_docs = "sphinx sphinx-automodapi sphinx_rtd_theme stsci_rtd_theme"
 def pip_packages_tests = "requests_mock ci_watson"
+def pip_index = "https://bytesalad.stsci.edu/artifactory/api/pypi/datb-pypi-virtual/simple"
+def pip_install_args = "--index-url ${pip_index} --progress-bar=off"
 
 // Generate distributions
 dist = new BuildConfig()
 dist.nodetype = 'linux'
 dist.name = 'dist'
-dist.conda_packages = ["numpy=${matrix_numpy[0]}"] + ["python=${matrix_python[0]}"]
+dist.conda_packages = ["python=${matrix_python[0]}"]
 dist.build_cmds = [
+    "pip install ${pip_install_args} numpy==${matrix_numpy[0]}",
+    "pip wheel ${pip_install_args} .",
     "python setup.py sdist",
-    "python setup.py bdist_egg",
-    "python setup.py bdist_wheel"
 ]
 matrix += dist
 
-
 // Compile documentation
 docs = new BuildConfig()
 docs.nodetype = 'linux'
 docs.name = 'docs'
-docs.conda_channels = ['http://ssb.stsci.edu/astroconda-dev']
-docs.conda_packages = conda_packages + conda_packages_docs + ["numpy=${matrix_numpy[0]}"] + ["python=${matrix_python[0]}"]
+docs.conda_packages = ["python=${matrix_python[0]}"]
 docs.build_cmds = [
-    "pip install -q ${pip_packages_docs}",
+    "sudo yum install -y graphviz texlive",
+    "pip install ${pip_install_args} numpy==${matrix_numpy[0]}",
+    "pip install ${pip_install_args} -r requirements-dev.txt --editable .",
+    "pip install ${pip_install_args} ${pip_packages_docs}",
     "python setup.py build_sphinx"
 ]
 matrix += docs
 
+// Generate pip build and test matrix
+for (python_ver in matrix_python) {
+    for (numpy_ver in matrix_numpy) {
+        for (astropy_ver in matrix_astropy) {
+            def name = "pip_py${python_ver}np${numpy_ver}ap${astropy_ver}"
+            bc = new BuildConfig()
+            bc.nodetype = 'linux'
+            bc.env_vars = test_env
+            bc.name = name
+            bc.conda_packages = ["python=${python_ver}"]
+            bc.build_cmds = [
+                "pip install ${pip_install_args} numpy==${matrix_numpy[0]}",
+                "pip install ${pip_install_args} ${pip_packages_tests}",
+                "pip install ${pip_install_args} -r requirements-dev.txt .",
```

Comment:
I think we need to change this line to
```
pip install ${pip_install_args} -r requirements-dev.txt .[test]
```
?

Reply:
And we probably don't need `pip_packages_tests` and `pip_packages_docs` defined or used anymore.